src/Microsoft.PowerShell.Commands.Utility/commands/utility/CSVCommands.cs

```diff
@@ -206,8 +206,27 @@ public SwitchParameter NoClobber
         /// Encoding optional flag
         /// </summary>
         [Parameter()]
-        [ValidateSetAttribute(new string[] { "Unicode", "UTF7", "UTF8", "ASCII", "UTF32", "BigEndianUnicode", "Default", "OEM" })]
-        public string Encoding { get; set; }
+        [ArgumentToEncodingTransformationAttribute()]
+        [ArgumentCompleter(typeof(EncodingArgumentCompleter))]
+        public Encoding Encoding
+        {
+            get
+            {
+                return _encoding;
+            }
+            set
+            {
+                if ( value == EncodingConversion.byteEncoding )
```

Comment:
These set methods need a comment as to why the comparison to EncodingConversion.byteEncoding is needed to provide a guide when creating new cmdlets that provide an Encoding parameter.

Reply:
Maybe add a helper, something like:
```c#
EncodingConversion.SetEncodingParameterHelper(ref _encoding, value);
```
Or:
```c#
_encoding = EncodingConversion.EncodingParameterHelper(value);
```


---
In reply to: [144073559](https://github.com/PowerShell/PowerShell/pull/5080#discussion_r144073559) [](ancestors = 144073559)